src/validation.pl

```diff
@@ -3,26 +3,70 @@
 :- use_module(library(debug)).
 :- use_module(library(files)).
 
-is_list([]).
-is_list([_|L]) :-
-    is_list(L).
+is_list_t(Ls, T) :-
+    match(Ls, [
+        arm(ground([]), (T = true)),
+        arm(composite([any | bind(Ls1)]), is_list_t(Ls1, T)),
+        arm(any, (T = false))
+    ]).
 
-is_list_t(L, true):-
-    is_list(L),!.
-
-is_list_t(_, false).
 
 file_exists_t(P, true):- file_exists(P),!.
 file_exists_t(_, false).
 
-is_license_t(license(name(N)), true):- is_list_t(N, true),!.
-is_license_t(license(name(_)), false).
+license_t_(license(name(N)), true):- is_list_t(N, true).
+license_t_(license(name(N)), false):- is_list_t(N, false).
+
+license_t_(license(name(N), path(P)), true):- is_list_t(N, true), is_list_t(P, true).
+license_t_(license(name(N), path(P)), false):- is_list_t(N, false), is_list_t(P, false).
+license_t_(license(name(N), path(P)), false):- is_list_t(N, true), is_list_t(P, false).
+license_t_(license(name(N), path(P)), false):- is_list_t(N, false), is_list_t(P, true).
```

Comment:
```suggestion
license_t_(license(name(N), path(P)), T) :-
    call((is_list_t(N), is_list_t(P)), T).
```

Remembering that `(',')/3` reifies conjunction, which is why you can write things like `(X = 1, Y = 2)` in a `if_/3` condition.